ClamavPlugin.inc.php

```diff
@@ -323,44 +327,63 @@ function clamscanHandleUpload($hookName, $args) {
 		//scan for viruses if file exists
 		if (null !== $file) {
 			$useSocket = $this->getSetting(CONTEXT_SITE, 'clamavUseSocket');
+			//Daemon
 			if ($useSocket === true) {
-				$message = $this->_clamDaemonFile($uploadedFile);
+				try {
+					$message = $this->_clamDaemonFile($uploadedFile);
+					//safe, continue with submission
+					if ($message == false) {
+						return false;
+					}
+				}
+				catch (ClamScanFailureException $e){
+					//couldn't scan, but ClamAV settings are permissive, continue with submission
+					if ( $this->getSetting(CONTEXT_SITE, 'allowUnscannedFiles')==="UNSCANNED_ALLOW") {
+						return false;
+					}
+				}
+			//clamscan executable	
 			} else {
-				$message = $this->_clamscanFile($uploadedFile);
-			}
-			//Virus found
-			if ($message == true) {
-				if ($message == 'timeout' && $this->getSetting(CONTEXT_SITE, 'allowUnscannedFiles')==="UNSCANNED_ALLOW"){
-					//allow file when timeout setting is permissive
-					return false;
+				try {
+					$message = $this->_clamscanFile($uploadedFile);
+					//safe, continue with submission
+					if ($message == false) {
```

Comment:
This should be `$message === false` to explicitly check the return value as literal false, rather than something falsey, like a virus named "0" or an empty string.